src/Orchard.Environment.Extensions/ExtensionManager.cs

```diff
@@ -256,6 +256,11 @@ public Task<ExtensionEntry> LoadExtensionAsync(IExtensionInfo extensionInfo)
 
         public async Task<IEnumerable<FeatureEntry>> LoadFeaturesAsync()
         {
+            // for testing
+            Parallel.ForEach(GetFeatures(),
```

Comment:
@sebastienros and @Jetski5822 

Just retried this part of code that i didn't really checked while working on the mvcless branch.

So tested by compiling quite all modules. First it was quite the same with both solutions, **around 65 s**. But in both cases i don't really see parallel compilation / loading.

Then by using the same pattern that seems to be right (or one of the followings) and just by replacing `LoadFeatureAsync()` with `LoadExtensionAsync()` to first trigger extensions compilation / loading, **it takes about 25 s in place of 65 s**. Note: Just for testing without try catch blocks.

            Parallel.ForEach(GetExtensions(),
                new ParallelOptions { MaxDegreeOfParallelism = 4 },
                    async e => await LoadExtensionAsync(e));

            Parallel.ForEach(GetExtensions(),
                new ParallelOptions { MaxDegreeOfParallelism = 4 },
                    e => LoadExtensionAsync(e));

           Parallel.ForEach(GetExtensions(), new ParallelOptions { MaxDegreeOfParallelism = 4 }, e =>
            {
                var extensionEntry = LoadExtensionAsync(e).Result;
            });

Maybe because `LoadFeatureAsync()` as an inner loop which load multiples extensions synchronously. So, maybe when entering in this loop, `Task.WhenAll()` just wait before evaluating another Task<TResult> object and then continuing the execution of the async method on another item, this where it was.